Scripts/Player/PlayerCollisionController.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerCollisionController : MonoBehaviour
+{
+
+    PlayerAnimationController animationController;
+    PlayerInventoryController inventory;
+    float time = 2;
+    private void Start()
+    {
+        animationController = GetComponent<PlayerAnimationController>();
+        inventory = GetComponent<PlayerInventoryController>();
+    }
+    private void OnTriggerEnter(Collider other)
+    {
+        {
+            if (other.gameObject.CompareTag("MedKit"))
+            {
+                inventory.MedkitCollected();//collecing medkit ++
+                Destroy(other.gameObject);//destroymedkit
+            }
+            if (other.gameObject.CompareTag("Bullet"))
+            {
+                Destroy(other.gameObject);
+                gameObject.GetComponent<PlayerManager>().OnGameover?.Invoke();
+            }
+        }
+    }
+    private void OnTriggerStay(Collider other)
+    {
+        if (other.gameObject.layer == 9 && !other.GetComponent<SoldierLifeStatus>().isAlive && inventory.Medkit > 0)//soldier
+        {
+            time -= Time.deltaTime;
+            if (time < 0)
+            {
+                other.GetComponent<SoldierLifeStatus>().health = 100;
+                inventory.UseMedkit();
+            }
+            animationController.ChangeAnimation(AnimationState.Healing);
```

Comment:
```suggestion
            animationController.CurrentAnimation = AnimationState.Healing;
```

public AnimationState CurrentAnimation { get {currentAnimation} set { //logic kontrolleri}}